internal/handlers/common/filter.go

```diff
@@ -594,76 +594,231 @@ func filterFieldExprSize(fieldValue any, sizeValue any) (bool, error) {
 
 // filterFieldExprBitsAllClear handles {field: {$bitsAllClear: value}} filter.
 func filterFieldExprBitsAllClear(fieldValue, maskValue any) (bool, error) {
-	fieldBinary, maskBinary, err := getBinaryParams(fieldValue, maskValue)
-	if err != nil {
-		return false, formatBitwiseOperatorErr(err, "$bitsAllClear", maskValue)
-	}
+	switch value := fieldValue.(type) {
+	case int32:
+		bitmask, err := getBinaryMaskParam(maskValue)
+		if err != nil {
+			return false, formatBitwiseOperatorErr(err, "$bitsAllClear", maskValue)
+		}
+		return (^uint64(value) & bitmask) == bitmask, nil
+
+	case int64:
+		bitmask, err := getBinaryMaskParam(maskValue)
+		if err != nil {
+			return false, formatBitwiseOperatorErr(err, "$bitsAllClear", maskValue)
+		}
+		return (^uint64(value) & bitmask) == bitmask, nil
 
-	for i := 0; i < len(fieldBinary.B); i++ {
-		if (fieldBinary.B[i] & maskBinary.B[i]) != 0 {
+	case float64:
+		// TODO check float negative zero
+		if value != math.Trunc(value) || math.IsNaN(value) || math.IsInf(value, 0) || value > math.MaxInt64 {
```

Comment:
can it be in several strings? 
why not to check negative zero now?

Reply:
Why do you want to split it?
Negative zero seems to be another tough thing. It's not so easy to check for that as far as I get.

Reply:
> Why do you want to split it? 

different conditions on separate lines.

> Negative zero seems to be another tough thing. It's not so easy to check for that as far as I get.

ok